test/httpClient.test.js

```diff
@@ -66,15 +66,16 @@ test('client calls a https server twice', (t) => {
 test('client calculates correct duration when using pipelining', (t) => {
   t.plan(4)
 
-  const lazyServer = helper.startServer({ delayResponse: 500 })
+  const delayResponse = 500
+  const lazyServer = helper.startServer({ delayResponse })
   const opts = lazyServer.address()
   opts.pipelining = 2
   const client = new Client(opts)
   let count = 0
 
   client.on('response', (statusCode, length, duration) => {
     t.equal(statusCode, 200, 'status code matches')
-    t.ok(duration > 500 && duration < 800)
+    t.ok(duration > delayResponse, `Expected response delay > ${delayResponse}ms but got ${duration}ms`)
```

Comment:
FYI the original reason for checking if duration < 800 was [this](https://github.com/mcollina/autocannon/pull/317#discussion_r544185281).

> our calculations were giving ~half results, so not checking if it takes less than 600ms might not catch if the calculations in future break and start giving 2x time. We can increase it to 700 if builds start to fail.

I know it wasn't the right fix and there should be a more predictable way to assert this but removing the upper limit check brings the risk I explained in the comment.

Reply:
@nherment could you handle this?

Reply:
Yes, I'll add a unit test for these calculations.

Reply:
@salmanm could you please point me to the 'calculations' you are referring to?

The code around the delayed response is pretty straightforward and I don't see anything that I would classify as calculations.

Reply:
Calcs were wrong for avg latency when using pipeline (See [this](https://github.com/mcollina/autocannon/issues/315)). Which was fixed by correctly calculating current expected response (`cer`) [here](https://github.com/mcollina/autocannon/pull/317/files#diff-9c6cf685b37a3aab04dbaf56e9a8d4bfb60041335227154d65f0d69d9c2f7b23R118).